http/client/request.go

```diff
@@ -81,10 +81,21 @@ func NewFileRequest(L *lua.LState) int {
 		return
 	}
 
-	if err := writeFile(files, writer); err != nil{
-		files.ForEach(func(k, v lua.LValue) {
-			writeFile(v.(*lua.LTable), writer)
-		})
+	var err error
+	if files.Len() == 0 {
+		err = writeFile(files, writer)
+	} else {
+		for key, value := files.Next(lua.LNil); key != lua.LNil; key, value = files.Next(key) {
```

Comment:
I still liked the `files.ForEach` style - was there some compelling reason to use the Next loop style?

Reply:
When an error occurs, the Next loop style can be easily interrupted, and Foreach cannot be interrupted.